eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -54,21 +83,22 @@ jobs:
         echo "##vso[task.prependpath]$(pwd)"
         echo "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT;]$(pwd)"
 
-    condition: contains(variables['vm.image'], 'macOS')
+    condition: and(contains(variables['vm.image'], 'macOS'), contains(variables['vcpkg.dep'], 'curl'))
```

Comment:
Instead of checking explicitly for `curl` we should skip the vcpkg steps where there are no `vcpkg.dep`s 

```
condition: and(succeeded(), contains(variables['vm.image']), 'macOS', not(eq(variables['vcpkg.dep'], '')))
```

[Reference docs on conditional functions](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#not)

Note the `succeeded()` call is in there because we need to evaluate whether previous steps have succeeded. This condition is implied when no condition is specified and needs to be re-inserted or these steps may run even when you cancel the pipeline.